pkg/manifests/manifests.go

```diff
@@ -3348,7 +3348,7 @@ func (f *Factory) HashSecret(secret *v1.Secret, data ...string) (*v1.Secret, err
 	}, nil
 }
 
-func serviceMonitors(fullServiceMonitor, minimalServiceMonitor func() (*monv1.ServiceMonitor, error)) ([]*monv1.ServiceMonitor, error) {
+func (f *Factory) serviceMonitors(fullServiceMonitor, minimalServiceMonitor func() (*monv1.ServiceMonitor, error)) ([]*monv1.ServiceMonitor, error) {
```

Comment:
```suggestion
func serviceMonitors(appendMinimal bool, fullServiceMonitor, minimalServiceMonitor func() (*monv1.ServiceMonitor, error)) ([]*monv1.ServiceMonitor, error) {
```
I'd probably argue to keep this is a pure function. `serviceMonitors` doesn't really need to know about the TP flag and we don't need ths recevier arg here for any purpose.